cypher-shell/src/main/java/org/neo4j/shell/state/BoltStateHandler.java

```diff
@@ -133,7 +133,20 @@ private void silentDisconnect() {
         }
     }
 
+    /**
+     * Reset the current session. This rolls back any open transactions.
+     */
     public void reset() {
-//        session.reset();
+        // TODO once drivers release next milestone
+        //if (session != null) {
+            // session.reset();
+        //}
+
+        // Clear current state
```

Comment:
The transaction methods are fine to keep. We need to set `tx` to null after reset. Doing it manually has the advantage of being testable. The internals of the bolt driver can't.